ale_linters/sql/sqlfluff.vim

```diff
@@ -19,7 +22,7 @@ function! ale_linters#sql#sqlfluff#Command(buffer) abort
     \    ale#Escape(l:executable)
     \    . ' lint'
 
-    let l:config_file = ale#path#FindNearestFile(a:buffer, '.sqlfluff')
+    let l:config_file = ale#path#FindNearestFile(a:buffer, ale#Var(a:buffer, 'sql_sqlfluff_config_file'))
```

Comment:
@w0rp tbh that would be much preferred. 

My original idea was to replace `%t` with `%s` in the [linter](https://github.com/dense-analysis/ale/blob/affd6029783b5611d63197650f959480c9211798/ale_linters/sql/sqlfluff.vim#L36) and [fixer](https://github.com/dense-analysis/ale/blob/affd6029783b5611d63197650f959480c9211798/autoload/ale/fixers/sqlfluff.vim#L24) and let the tool determines which configuration to load. The reason why it didn't load was because the linter/fixer was pointed to a temporary file and it couldn't find any config files. 

I just recently migrated to ale from null-ls and not the author of the original code,  so I wasn't sure if replacing the command with `%s` for both linter and fixer will have unintended consequences so I tried to make the code to be as backward compatible as possible.

Reply:
You might be able to get it to work by adding `'cwd': '%s:h'` to the linter config. You can look at the `rstcheck.vim` file for an example.

Reply:
> You might be able to get it to work by adding `'cwd': '%s:h'` to the linter config. You can look at the `rstcheck.vim` file for an example.

I've tried adding this, but doesn't really help in my case; e.g., my directory structure would be something like

```
repo  # Project working directory
├── pyproject.toml  # The config I want applied
└── queries
    └── query.sql  # The file I'm editing
```

So, if I start vim from the project root (in this case, `repo`), open `queries/query.sql`, then `cwd` for the linter is set to `repo/queires` instead of `repo`, and `sqlfluff` fails to find `pyproject.toml`; i.e., the lint command is

```
['bash', '-c', 'cd ''/repo/queries'' && ''sqlfluff'' lint ...]
```

instead of the desired

```
['bash', '-c', 'cd ''/repo'' && ''sqlfluff'' lint ...]
```

If I run `sqlfluff` from `vim` directly (e.g., `:!sqlfluff lint %`) it _does_ find the correct config; so it looks like `vim` "knows" the correct `cwd`, but I couldn't find any format strings in `ale-command-format-strings` that would give me that path.